c++/src/kj/common.h

```diff
@@ -1377,6 +1377,46 @@ class Maybe {
     }
   }
 
+  template <typename F,
+      typename Result = decltype(instance<bool>() ? instance<T&>() : instance<F>()())>
+  Result orDefault(F&& lazyDefaultValue) & {
+    if (ptr == nullptr) {
+      return lazyDefaultValue();
+    } else {
+      return *ptr;
+    }
+  }
+
+  template <typename F,
+      typename Result = decltype(instance<bool>() ? instance<const T&>() : instance<F>()())>
+  const Result orDefault(F&& lazyDefaultValue) const & {
```

Comment:
I think `Result` will already be derived to include `const` where needed, and declaring it `const` again here doesn't mean anything.